### PR TITLE
nixos/tests/vscodium-wayland: init

### PR DESCRIPTION
**Pull request by Synthetica9**
###### Motivation for this change

I couldn't find the issue where this was originally mentioned, but here's a test for vscodium on wayland. It might be nice to unify the test scripts for wayland and xorg, but I don't really see a way how (suggestions welcome!)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Synthetica9** [comment]
Related: https://github.com/NixOS/nixpkgs/pull/123680

cc @primeos

**Comment by Mindavi** [comment]
Offtopic: @Synthetica9 it seems like your comment refers to this very PR, I guess that was not intended?

Somehow imagemagick is rebuilding, but I did a nixpkgs-review and that seemed to pass... Not sure what's happening there. Otherwise, looks good to me. This seems like a good test of the basic functionality.

**Comment by Synthetica9** [comment]
Converted this to draft again, currently merging the two tests. 

> Offtopic: @Synthetica9 it seems like your comment refers to this very PR, I guess that was not intended?

No, I meant to refer to https://github.com/NixOS/nixpkgs/pull/118360 and messed up the copy-paste.

**Comment by Mindavi** [comment]
I might've been running these tests incorrectly before. How do I run these? Running `nix-build nixos/tests/vscodium.nix` gives errors about the paths, so I guess I should run them in a different way.

**Comment by Synthetica9** [comment]
Oh, my fault! Try again with this one!

Easiest way to test is `nix-build -A nixosTests.vscodium`

**Review by Mindavi — APPROVED** [review]
LGTM and runs on my machine. Did take ~230s to run everything, but that might be related to me building all kinds of things at the same time.

**Comment by Synthetica9** [comment]
still relevant

**Comment by Mindavi** [comment]
Ran this once again and now it takes _only_ 150 seconds.

I had trouble merging master / rebasing master on top of this, so haven't tested with tat.

**Comment by Synthetica9** [comment]
> Ran this once again and now it takes _only_ 150 seconds.
> 
> I had trouble merging master / rebasing master on top of this, so haven't tested with that. I guess that needs to be solved though.

tbf if you've used the old branch nothing has changed... So any speed gained or lost is either your hardware or just fluctuations.

I've rebased on master and made it a bit better, probably gonna merge it if there are no strong objections. @turion as maintainer of the test?

File: nixos/tests/vscodium.nix [hunk 1]
```diff
@@ -1,47 +1,64 @@
...
+        test_string = 'testfile'
 
-    # Start VSCodium with a file that doesn't exist yet
-    machine.fail("ls /home/alice/foo.txt")
-    machine.succeed("su - alice -c 'codium foo.txt' >&2 &")
+        machine.send_key('ctrl-n')
+        machine.wait_for_text('Untitled')
+        machine.screenshot('empty_editor')
 
-    # Wait for the window to appear
```
Comment (turion) [inline nixos/tests/vscodium.nix:36]: I think you could keep the comments in your improved version of the test.

**Review by turion — APPROVED** [review]
This is great! Just a minor comment.

**Pull request merged**